### PR TITLE
Handle jsonl

### PR DESCRIPTION
**Pull request by jgindin**
# Description

**Multi-Format Layout Parser (JSON, JSONL, Single-Object)**

* Enhances tryParseJsonArray in utils/json.ts to parse standard JSON arrays ([...]), single JSON objects ({...} → [obj]), and streaming JSON Lines ({...}\n{...}).
* Updates RawFrame and ChatCleaner to use unified JSONL-tolerant parsing and layout snapshot validation.
* Adds comprehensive unit tests covering single objects, multi-line JSONL streams, and malformed edge cases.

## Pre-launch Checklist

- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I read the [Style Guide].
- [ ] I have added updates to the [CHANGELOG].
- [ ] I updated/added relevant documentation.
- [x] My code changes (if any) have tests.
- [ ] If my branch is on fork, I have verified that [scripts/e2e_test.sh](../scripts/e2e_test.sh) passes.

If you need help, consider asking for advice on the [discussion board].

### CODE REVIEW

File: shell/src/app/utils/json.ts [hunk 1]
```diff
@@ -23,16 +23,61 @@
...
+  } else if (trimmed.startsWith('{') && trimmed.endsWith('}')) {
+    try {
+      const parsed = JSON.parse(trimmed);
+      if (parsed !== null && typeof parsed === 'object' && !Array.isArray(parsed)) {
+        return [parsed];
+      }
+    } catch (e) {
+      // Ignore object parse errors
     }
   }
```
[the remark this answers is not in the document]
Comment (jgindin) [reply shell/src/app/utils/json.ts]: fixed

File: shell/src/app/chat/a2ui-payload-parser/a2ui-payload-parser.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  }
+
+  return {blocks: parsedBlocks, wasHealed};
+}
+
+/**
+ * Attempts simple token-healing heuristics on a malformed JSON string.
+ */
+export function attemptSyntaxHealing(line: string): unknown | null {
+  let patched = line.trim();
```
[the remark this answers is not in the document]
Comment (jgindin) [reply shell/src/app/chat/a2ui-payload-parser/a2ui-payload-parser.ts]: fixed

File: shell/src/app/utils/json.spec.ts [hunk 3]
```diff
@@ -24,12 +24,29 @@ describe('JSON Array Parser Utilities', () => {
...
     expect(tryParseJsonArray('123')).toBeNull();
+    expect(tryParseJsonArray('{"a": 1}\n{"b": 2}\ninvalid\n{"c": 3}')).toBeNull();
+    expect(tryParseJsonArray('{"a": 1\n{"b": 2}')).toBeNull();
+    expect(tryParseJsonArray('{"a": 1}\n{ "syntax_error": }')).toBeNull();
+  });
+
+  it('parses a single JSON object as a single-element array', () => {
+    expect(tryParseJsonArray('{"not": "an array"}')).toEqual([{not: 'an array'}]);
+    expect(tryParseJsonArray('  {"foo": "bar"}  ')).toEqual([{foo: 'bar'}]);
+  });
```
[the remark this answers is not in the document]
Comment (jgindin) [reply shell/src/app/utils/json.spec.ts:47]: fixed

File: shell/src/app/chat/a2ui-payload-parser/a2ui-payload-parser.ts [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    const parsedSingle = JSON.parse(content);
+    if (Array.isArray(parsedSingle)) {
+      return {blocks: parsedSingle, wasHealed};
+    }
+    if (parsedSingle && typeof parsedSingle === 'object') {
+      return {blocks: [parsedSingle], wasHealed};
+    }
+  } catch {
+    // Continue to line-by-line healing
+  }
```
[the remark this answers is not in the document]
Comment (jgindin) [reply shell/src/app/chat/a2ui-payload-parser/a2ui-payload-parser.ts]: fixed

**Review by jonsharkey — APPROVED** [review]
